### ISSUE TITLE
Loading IPy Extensions upon kernel start - Where has c.InteractiveShellApp.extensions gone ?

### ISSUE THREAD

**Issue by oscar6echo** [jupyter/notebook#903]
I don't know if this is the right place to ask. If not please disregard and close. I would be grateful for some directions though.

A number of seemly well informed posts (see below) mention how to load extensions by default, with the `c.InteractiveShellApp.extensions` option in the notebook config file, so that the  `%load_ext <ext_name>`commands are not necessary.
The default jupyter_notebook_config.py does not list `c.InteractiveShellApp.extensions` any longer.
The documentation says the same: http://jdfreder-notebook.readthedocs.org/en/latest/config.html

So how to do that ?
Even temporarily as I understand everything is in flux.

Blog posts:
https://mindtrove.info/#ipy-kernel-exts
http://litaotao.github.io/ipython-learning/

(PS: For notebook js extensions, everything works fine)

**Comment by willingc** [comment]
@oscar6echo While I am not 100% sure on the answer to your question (pinging @jdfreder @ellisonbg @Carreau for additional insights), I believe that you may still add the `c.InteractiveShellApp.extensions` option to the notebook config file even though it is no longer bundled in the config file by default. 

Here are the links to current documentation related to IPython extensions in Jupyter and IPython docs:
[Jupyter documentation on migration of extensions](https://jupyter.readthedocs.org/en/latest/migrating.html#notebook-extensions)
[Documentation of IPython extensions](https://ipython.readthedocs.org/en/latest/config/extensions/)
[Bundled IPython extensions](https://ipython.readthedocs.org/en/latest/config/extensions/#extensions-bundled-with-ipython)
[How to install and add to configuration file an IPython Extension]
(https://ipython.readthedocs.org/en/latest/config/extensions/#ipython-extensions)

**Comment by ellisonbg** [comment]
I don't think that the IPython kernel will read the
jupyter_notebook_config.py file. You would have to put that in your ipython
config file, which is loaded by all ipython kernels. Think of it this way -
jupyter is language independent and doesn't know anything about how
particular kernels in different languages are configured. Does that make
help?

On Mon, Dec 28, 2015 at 4:08 PM, Carol Willing <notifications@github.com>
wrote:

> @oscar6echo <https://github.com/oscar6echo> While I am not 100% sure on
> the answer to your question (pinging @jdfreder
> <https://github.com/jdfreder> @ellisonbg <https://github.com/ellisonbg>
> @Carreau <https://github.com/Carreau> for additional insights), I believe
> that you may still add the c.InteractiveShellApp.extensions option to the
> notebook config file even though it is no longer bundled in the config file
> by default.
>
> Here are the links to current documentation related to IPython extensions
> in Jupyter and IPython docs:
> Jupyter documentation on migration of extensions
> <https://jupyter.readthedocs.org/en/latest/migrating.html#notebook-extensions>
> Documentation of IPython extensions
> <https://ipython.readthedocs.org/en/latest/config/extensions/>
> Bundled IPython extensions
> <https://ipython.readthedocs.org/en/latest/config/extensions/#extensions-bundled-with-ipython>
> How to install and add to configuration file an IPython Extension
> <https://ipython.readthedocs.org/en/latest/config/extensions/#ipython-extensions>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/issues/903#issuecomment-167682014>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by oscar6echo** [comment]
Yes it makes help and sense and it works.
Thanks to both of you.

For the record, here is what I did:

1/
Erase just about every file (not folder) in my .ipython directory for fear some remnant files from before the Big Split cause trouble.
2/
Run from terminal `ipython notebook --generate-config`. It failed until I re-created my default profile with ` ipython profile create`. Then it worked. The `ipython notebook --generate-config` command created 2 config files: 
+ .ipython/profile_default/ipython_config.py
+ .ipython/profile_default/ipython_kernel_config.py  

3/
Add `c.InteractiveShellApp.extensions = [ 'version_information']`to the first one. Putting this snippet in the second one, or not, did not seem to have an impact.

4/ Re-start the kernel and the '%version_information' magic (http://nbviewer.ipython.org/github/jrjohansson/version_information/blob/master/example.ipynb) I used was listed below `%lsmagic` and indeed available through `%version_information`.

5/ My config:
`````
$ conda list  | grep -E 'jupyter|ipython' 
ipython                   4.0.1                    py27_0    defaults
ipython-genutils          0.1.0                     <pip>
ipython_genutils          0.1.0                    py27_0    defaults
jupyter                   1.0.0                    py27_1    defaults
jupyter_client            4.1.1                    py27_0    defaults
jupyter_console           4.0.3                    py27_0    defaults
jupyter_core              4.0.6                    py27_0    defaults
````
By the way, I noticed that the folder 'extensions' in .ipython (surely a remnant) is empty and could be deleted without apparent consequences.

(And congrats to all the people working on IPython & Jupyter - Very very useful !)
